Mathlib/RingTheory/TensorProduct/MvPolynomial.lean

```diff
@@ -186,6 +186,44 @@ noncomputable def scalarRTensorAlgEquiv :
     MvPolynomial σ R ⊗[R] N ≃ₐ[R] MvPolynomial σ N :=
   rTensorAlgEquiv.trans (mapAlgEquiv σ (Algebra.TensorProduct.lid R N))
 
+variable (R)
+variable (A : Type*) [CommRing A] [Algebra R A]
+
+/-- Tensoring `MvPolynomial σ R` on the left by an `R`-algebra `A` is algebraically
+equivalent to `M̀vPolynomial σ A`. -/
+noncomputable def algebraTensorAlgEquiv :
+    A ⊗[R] MvPolynomial σ R ≃ₐ[A] MvPolynomial σ A := AlgEquiv.ofAlgHom
```

Comment:
Apparently this already [exists](https://github.com/leanprover-community/mathlib4/pull/12293/files#diff-fcd41314736c9eb63afb8cedfaae8800b725b0f3003d437246bf1c6caf4b3f81R185-R186) thanks to @AntoineChambert-Loir

Reply:
I thought so as well, but I don't think that is entirely correct. First, in the equiv you cited `N` is just a module, so the equiv is only `R`-linear but I need `A`-linearity. Secondly, even if you ask `N` to be an `R`-algebra, the LHS in the equiv you cited is "the wrong way" round for my application: I think @AntoineChambert-Loir had in mind to enhance it to something `MvPolynomial σ R`-linear for which `MvPolynomial σ R` has to be on the left of the `⊗[R]`. But I need it to be `A`-linear, so `A` has to be on the left of `⊗[R]`.

Reply:
And while I can probably massage `scalarRTensorAlgEquiv` into the thing I want by using `Algebra.TensorProduct.comm` and promoting it to an `A`-linear map, I think this would be longer and would have worse definitional properties than the direct definition.

Reply:
`N` is a CommSemiring there (that's a prerequisite for MvPolynomial), but yes I neglected the A-linearity, which requires A on the left. Can you please generalize from Ring to Semiring though?

Reply:
Done

Reply:
Thanks, LGTM!